cheatsheets_draft/Multifactor_Authentication_Cheat_Sheet.md

```diff
@@ -0,0 +1,327 @@
+
+- Types of MFA that require users to have specific hardware can introduce significant costs and administrative overheads.
+- Users may become locked out of their accounts if they lose or are unable to use their other factors.
+- MFA introduces additional complexity into the application.
+- Many MFA solutions add external dependencies to systems, which can introduce security vulnerabilities or single points of failure.
+- Processes implemented to allow users to bypass or reset MFA may be exploitable by attackers.
+- Requiring MFA may prevent some users from accessing the application.
+
+# Quick Recommendations
+
+Exactly when and how MFA is implemented in an application will vary on a number of different factors, including the threat model of the application, the technical level of the users, and the level of administrative control over the users. These need to be considered on a per-application basis.
+
+However, the following recommendations are generally appropriate for most applications, and provide an initial starting point to consider.
+
+- Provide the option for users to enable MFA on their accounts using [TOTP](#software-otp-tokens).
+- Require MFA for administrative or other high privileged users.
+- Consider whitelisting corporate IP ranges so that MFA is not required from them.
+- Allow the user to remember the use of MFA in their browser, so they are not prompted every time they login.
+- Implement a secure process to allow users to reset their MFA.
+
+# Implementing MFA
+
+## When to Require MFA
+
+The most important place to require MFA on an application is when the user logs in. However, depending on the functionality available, it may also be appropriate to require MFA for performing sensitive actions, such as:
+
+- Changing passwords or security questions.
+- Changing the email address associated with the account.
+- Disabling MFA.
+- Elevating a user session to an administrative session.
+
+If the application provides multiple ways for a user to authenticate these should all require MFA, or have other protections implemented. A common area that is missed is if the application provides a separate API that can be used to login, or has an associated mobile application.
+
+## Improving Usability
+
+Having to frequently login with MFA creates an additional burden for users, and may cause them to disable MFA on the application. A number of mechanisms can be used to try and reduce the level of annoyance that MFA causes. However, these types of measures do decrease the security provided by MFA, so need to be risk assessed to find a reasonable balance of security and usability for the application.
+
+- Remembering the user's browser so they don't need to use MFA every time.
+  - This can either be permanent, or for a period of a few days.
+  - This needs to be done with more than just a cookie, which could be stolen by an attacker.
```

Comment:
What would be the other mean, other than a cookie?

Reply:
I was thinking a cookie + IP address pair. Updated in bf6a9e0422abfc818fcf91f0ed7b13d38cefdecd.